src/Dictionaries/DictionaryStructure.cpp

```diff
@@ -376,6 +377,14 @@ std::vector<DictionaryAttribute> DictionaryStructure::getAttributes(
         if ((range_min && name == range_min->name) || (range_max && name == range_max->name))
             continue;
 
+        if (attribute_names.find(name) != attribute_names.end())
```

Comment:
Insert method returns a bool - if the item was inserted.
No need for duplicate find/insert.

Reply:
Fixed.